homeassistant/components/russound_rio/config_flow.py

```diff
@@ -0,0 +1,93 @@
+"""Config flow to configure russound_rio component."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from russound_rio import Russound
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN, RUSSOUND_RIO_EXCEPTIONS
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=9621): cv.port,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Russound RIO configuration flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize workflow."""
+        self._host: str = ""
+        self._name: str = ""
+        self._port: int = 9621
+        self._entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA, errors=errors
+            )
+
+        self._host = user_input[CONF_HOST]
+        self._name = user_input[CONF_NAME]
+        self._port = user_input[CONF_PORT]
+        return await self.async_step_pairing()
+
+    async def async_step_pairing(
```

Comment:
If we are not doing anything special here, let's combine the two flows as we usually do.
so:
```py
if user_input is not None:
    ....the connection check + create entry
show form (which will be shown if the connection checks fails)
```

Reply:
The intention was to have async_step_pairing() as a common call for both the `user` step and `import` step to reduce duplicate code. My other two solutions would be:
1) Duplicating connection logic (or making separate connection logic, which is what the pairing step represents) between the user and import steps.
2) Combining user and pairing step into one, and calling user step from data passed from import step. I thought this might be visually confusing, so went with the former.

Reply:
Oh I missed that this was also used in the import step. I personally always like to duplicate it, because it will be removed eventually, and you have to abort the flow in the case of an import flow instead of returning errors with the form.

Reply:
Yep, I did the abort to be able to combine the logic. If returning the errors is the expected behavior for the user step, I'll separate the logic.

Reply:
Logic has been separated.